starlite/asgi/routing_trie/types.py

```diff
@@ -69,25 +72,49 @@ class RouteTrieNode:
     """
     Designate the node as being a mount route.
     """
+    path: str
+    """
+    String path to this node.
+    """
     path_parameters: List["PathParameterDefinition"]
     """
     A list of tuples containing path parameter definitions. This is used for parsing extracted path parameter values.
     """
 
 
-def create_node() -> RouteTrieNode:
+def create_node(
+    parent_node: Optional[RouteTrieNode] = None, component: Optional[Union[str, PathParameterDefinition]] = None
+) -> RouteTrieNode:
     """Create a RouteMapNode instance.
 
+    Uses `parent_node` and `component` to construct the `URL` path of the node. If neither provided, path of node is
+    `'/'`.
+
+    If `parent_node` and `component` provided, adds the new node to the parent's `children` mapping.
+
+    Args:
+        parent_node: The parent of the new node, if not root.
+        component: The path component that the node is to represent, or `None` if root.
+
     Returns:
         A route map node instance.
     """
+    parent_path = ("" if parent_node is None else parent_node.path).rstrip("/")
+    node_path = f"{{{component.full}}}" if isinstance(component, PathParameterDefinition) else component or ""
+    path = "/".join((parent_path, node_path))
```

Comment:
I added the path of the current node so that I could use it to generate an error message but it might be useful elsewhere also.